api/envoy/extensions/health_checkers/cached/v3/cached.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package envoy.extensions.health_checkers.cached.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.health_checkers.cached.v3";
+option java_outer_classname = "CachedProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/health_checkers/cached/v3;cachedv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Cached]
+// Cached health checker :ref:`configuration overview <config_health_checkers_cached>`.
+// [#extension: envoy.health_checkers.cached]
+
+// [#next-free-field: 9]
+message Cached {
```

Comment:
Can/should the cached server be an upstream cluster?
This should take care of all the connection parameters.

Reply:
No, this parameters are dedicated to a redis cache server
Which is handled by the implementation of this PR.

https://github.com/envoyproxy/envoy/blob/badad5f2aadde7523be4e422146ff10a099055db/source/extensions/health_checkers/cached/hiredis.cc